### PR TITLE
Add spell check workflow

### PR DESCRIPTION
**Pull request by hezhizhen**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

Add https://github.com/crate-ci/typos/ to workflow for spell check and fix existing typos.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/apis/v1alpha1/kwokctl_configuration_types.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ type KwokctlConfigurationOptions struct {
 	KubeSchedulerImage string `json:"kubeSchedulerImage,omitempty"`
 
 	// KwokControllerImage is the image of Kwok.
-	// is the default value for flag --contoller-image and env KWOK_CONTROLLER_IMAGE
+	// is the default value for flag --controller-image and env KWOK_CONTROLLER_IMAGE
```
Comment (wzshiming) [inline pkg/apis/v1alpha1/kwokctl_configuration_types.go:146]: I think these 2 typo fixes can be submitted separately.

File: .github/workflows/typos.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+name: Spell Check
```
Comment (wzshiming) [inline .github/workflows/typos.yaml:1]: There are no plans to introduce this

Comment (hezhizhen) [reply .github/workflows/typos.yaml:1]: May I know why? I think this workflow can help reduce typo automatically.

Comment (wzshiming) [reply .github/workflows/typos.yaml:1]: It's a little low on users at the moment, and we'll consider introducing it when it gets more users, or when the Kubernetes main repository uses it.

Comment (hezhizhen) [reply .github/workflows/typos.yaml:1]: I see. I'll close this PR and create a new PR to fix typo.

Comment (hezhizhen) [reply .github/workflows/typos.yaml:1]: https://github.com/kubernetes-sigs/kwok/pull/317

File: pkg/apis/v1alpha1/kwokctl_configuration_types.go (hunk 1 above)
Comment (hezhizhen) [reply pkg/apis/v1alpha1/kwokctl_configuration_types.go:146]: https://github.com/kubernetes-sigs/kwok/pull/317

**Pull request closed without merging**